test/Sentry.Tests/Protocol/TransactionTests.cs

```diff
@@ -526,4 +562,70 @@ public void ISpan_GetTransaction_FromSpan()
         // Assert
         Assert.Same(transaction, result);
     }
+
+    [Fact]
+    public async Task Idle_transaction_should_finish_within_idle_timeout_specified()
+    {
+        // Arrange
+        var client = Substitute.For<ISentryClient>();
+        var options = new SentryOptions
+        {
+            Dsn = ValidDsn,
+            IdleTimeout = TimeSpan.FromMilliseconds(2)
+        };
+        var hub = new Hub(options, client);
+        var context = new TransactionContext(
+            SpanId.Create(),
+            SpanId.Create(),
+            SentryId.Create(),
+            "my name",
+            "my operation",
+            "description",
+            SpanStatus.Ok,
+            null,
+            true,
+            TransactionNameSource.Component
+        );
+
+        var transaction = new TransactionTracer(hub, context);
+
+        // Act
+        await Task.Delay(20);
```

Comment:
@SeanFeldman [this stack overflow](https://stackoverflow.com/questions/31742521/accuracy-of-task-delay) might be relevant here.

Reply:
Validation based on time is tricky. I'm not sure `Thread.Sleep()` needs to replace `Task.Delay()`, @jamescrosswell.

Looking at the [SO answer by Stephen Cleary](https://stackoverflow.com/a/20084603/59563), I need a logical delay, not block the current thread. I will try to increase the delay from milliseconds to seconds to see if the issue is with such a low wait time.

Reply:
Bumping time has worked. Can't use ms.